packages/marshal/test/test-encodePassable.js

```diff
@@ -0,0 +1,156 @@
+
+const r2e = new Map();
+const e2r = [];
+
+const encodeThing = (prefix, r) => {
+  if (r2e.has(r)) {
+    return r2e.get(r);
+  }
+  const result = `${prefix}${e2r.length}`;
+  r2e.set(r, result);
+  e2r.push(r);
+  return result;
+};
+
+const decodeThing = (prefix, e) => {
+  assert(e.startsWith(prefix), X`unexpected encoding ${e}`);
+  const i = Number(BigInt(e.substring(1)));
+  assert(i >= 0 && i < e2r.length);
+  return e2r[i];
+};
+
+const compareRemotables = (x, y) =>
+  compareRank(encodeThing('r', x), encodeThing('r', y));
+
+const encodePassable = makeEncodePassable({
+  encodeRemotable: r => encodeThing('r', r),
+  encodePromise: p => encodeThing('?', p),
+  encodeError: er => encodeThing('!', er),
+});
+
+const decodePassable = makeDecodePassable({
+  decodeRemotable: e => decodeThing('r', e),
+  decodePromise: e => decodeThing('?', e),
+  decodeError: e => decodeThing('!', e),
+});
+
+const { comparator: compareFull } = makeComparatorKit(compareRemotables);
+
+const asNumber = new Float64Array(1);
+const asBits = new BigUint64Array(asNumber.buffer);
+const getNaN = (hexEncoding = '0008000000000000') => {
+  let bits = BigInt(`0x${hexEncoding}`);
+  bits |= 0x7ff0000000000000n;
+  if (!(bits & 0x0001111111111111n)) {
+    bits |= 0x0008000000000000n;
+  }
+  asBits[0] = bits;
+  return asNumber[0];
+};
+
+const NegativeNaN = getNaN('ffffffffffffffff');
+
+/** @type {[Key, string][]} */
+const goldenPairs = harden([
+  [1, 'fbff0000000000000'],
+  [-1, 'f400fffffffffffff'],
+  [NaN, 'ffff8000000000000'],
+  [NegativeNaN, 'ffff8000000000000'],
+  [0, 'f8000000000000000'],
+  [Infinity, 'ffff0000000000000'],
+  [-Infinity, 'f000fffffffffffff'],
+  [-1234567890n, 'n#90:8765432110'],
+  [-123456789n, 'n1:876543211'],
+  [-1000n, 'n6:9000'],
+  [-999n, 'n7:001'],
+  [-1n, 'n9:9'],
+  [-0n, 'p1:0'],
+  [37n, 'p2:37'],
+  [123456789n, 'p9:123456789'],
+  [1234567890n, 'p~10:1234567890'],
+  [934857932847598725662n, 'p~21:934857932847598725662'],
+]);
+
+test('golden round trips', t => {
+  for (const [k, e] of goldenPairs) {
+    t.is(encodePassable(k), e, 'does k encode as expected');
+    t.is(decodePassable(e), k, 'does the key round trip through the encoding');
+  }
+  // Not round trips
+  t.is(encodePassable(-0), 'f8000000000000000');
+  t.is(decodePassable('f0000000000000000'), NaN);
+});
+
+const orderInvariants = (t, x, y) => {
+  const rankComp = compareRank(x, y);
+  const fullComp = compareFull(x, y);
+  if (rankComp !== 0) {
+    t.is(rankComp, fullComp);
+  }
+  if (fullComp === 0) {
+    t.is(rankComp, 0);
+  } else {
+    t.assert(rankComp === 0 || rankComp === fullComp);
+  }
+  const ex = encodePassable(x);
+  const ey = encodePassable(y);
+  const encComp = compareRank(ex, ey);
+  if (fullComp !== 0) {
+    t.is(encComp, fullComp);
+  }
+};
+
+test('order invariants', t => {
+  for (let i = 0; i < sample.length; i += 1) {
+    for (let j = i; j < sample.length; j += 1) {
+      orderInvariants(t, sample[i], sample[j]);
```

Comment:
I postponed addressing https://github.com/endojs/endo/pull/1260#discussion_r1003711830 from #1290 to this PR, since it was about usage of fast-check. So this is a note to address it here.

Reply:
At [xxx](https://github.com/endojs/endo/pull/1260#discussion_r1003711830) @gibson042 wrote

> Shouldn't this be structured like `await fc.assert(fc.property(…))` rather than using a direct list?

Reply:
In discussion, we realized this suggestion is already covered by the last test case. However, this non-fuzzing version still serves a purpose. It ensures that at least these always get tested. Moved and added explanation. Resolving.